File: Basics/While-Loops/Lab/moving.py
free_room_width = int(input())
free_room_length = int(input())
free_room_height = int(input())
free_room_area = free_room_width * free_room_length * free_room_height
is_room_space_end = False
current_boxes_count = input()

while current_boxes_count != "Done":
    current_boxes_count = int(current_boxes_count)

    if free_room_area <= current_boxes_count:
        is_room_space_end = True
        break

    free_room_area -= current_boxes_count
    current_boxes_count = input()

if is_room_space_end:
    print(f"No more free space! You need {abs(free_room_area - current_boxes_count)} Cubic meters more.")
else:
    print(f"{free_room_area} Cubic meters left.")

File: JavaScript/Basics/0.5-While-Loops/Lab/moving.js
function moving(input) {
    let w = Number(input.shift());
    let l = Number(input.shift());
    let h = Number(input.shift());
    let free_space = w * l * h; let i = 0;
    while (input[i] != "Done") {
        let box = Number(input[i]);
        free_space -= box;
        if (free_space < 0) {
            return `No more free space! You need ${Math.abs(free_space)} Cubic meters more.`
        }
        i++;
    }
    return `${free_space} Cubic meters left.`;
}


console.log(moving(["10",
"10",
"2",
"20",
"20",
"20",
"20",
"122"]));
console.log(moving(["10",
"1",
"2",
"4",
"6",
"Done"]));
